cfme/tests/openstack/cloud/test_networks.py

```diff
@@ -208,20 +218,24 @@ def test_add_gateway_to_router(router, ext_subnet):
                 ext_network_subnet=ext_subnet.name)
     wait_for(router.provider_obj.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600,
              delay=10)
-    router.browser.refresh()
+    wait_for(lambda: router.cloud_network == ext_subnet.network,
+             delay=15, timeout=600, fail_func=router.browser.refresh)
     assert router.cloud_network == ext_subnet.network
 
 
 def test_add_interface_to_router(router, subnet):
     """Adds interface (subnet) to router"""
+    view = navigate_to(router, 'Details')
+    subnets_count_before_adding = int(view.entities.relationships.get_text_of('Cloud Subnets'))
     router.add_interface(subnet.name)
-    wait_for(router.provider_obj.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600,
-             delay=10)
-    router.browser.refresh()
+    wait_for(router.provider_obj.is_refreshed, func_kwargs=dict(refresh_delta=20), timeout=800,
+             delay=30)
     # TODO: verify the exact entities' names and relationships, not only count
-    view = navigate_to(router, 'Details')
-    subnets_count = int(view.entities.relationships.get_text_of('Cloud Subnets'))
-    assert subnets_count == 1  # Compare to '1' because clean router was used initially
+    wait_for(lambda: int(view.entities.relationships.get_text_of('Cloud Subnets')) ==
+             (subnets_count_before_adding + 1),
+             delay=30, timeout=800, fail_func=router.browser.refresh)
+    subnets_count_after_adding = int(view.entities.relationships.get_text_of('Cloud Subnets'))
```

Comment:
Required:
@dkholodo: No need to capture `subnets_count_after_adding` after `wait_for`. If `wait_for` fails just capture the exception and `assert` `False`... like
```
try:
    wait_for(
        lambda: int(view.entities.relationships.get_text_of("Cloud Subnets"))
        == (subnets_count_before_adding + 1),
        delay=30,
        timeout=800,
        fail_func=router.browser.refresh,
    )
except TimedOutError:
    assert False, "Some expaination about failuers"
```

Reply:
@digitronik Fixed